File: android-room-sample-master/app/src/javaCommon/java/com/emreeran/android/roomsample/db/vo/PostWithUser.java
package com.emreeran.android.roomsample.db.vo;

import android.arch.persistence.room.Embedded;

import com.emreeran.android.roomsample.db.entity.Post;
import com.emreeran.android.roomsample.db.entity.User;

import java.util.Objects;

/**
 * Created by Emre Eran on 9.05.2018.
 */
public class PostWithUser {
    @Embedded
    public final Post post;

    @Embedded(prefix = "user_")
    public final User user;

    public PostWithUser(Post post, User user) {
        this.post = post;
        this.user = user;
    }

    @Override
    public String toString() {
        return "PostWithUser{" +
                "postWithUser=" + post +
                ", user=" + user +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PostWithUser that = (PostWithUser) o;
        return Objects.equals(post, that.post) &&
                Objects.equals(user, that.user);
    }

    @Override
    public int hashCode() {
        return Objects.hash(post, user);
    }
}

File: android-room-sample-master/app/src/kotlinCommon/java/com/emreeran/android/roomsample/db/vo/PostWithUser.kt
package com.emreeran.android.roomsample.db.vo

import android.arch.persistence.room.Embedded
import com.emreeran.android.roomsample.db.entity.Post
import com.emreeran.android.roomsample.db.entity.User

/**
 * Created by Emre Eran on 26.05.2018.
 */
data class PostWithUser(
        @Embedded val post: Post,
        @Embedded(prefix = "user_") val user: User
)